### PR TITLE
Next

### PR DESCRIPTION
**Pull request by thingsplode**
*Issue #, if available:*

*Description of changes:*
New unit tests
Simplified configuration 
Typos and pythonic modifications

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: driver/aws/datalake_api.py [hunk 1]
```diff
@@ -22,42 +22,30 @@ def __init__(self, path_key):
...
     s3 = providers.get_s3()
-    rsp = s3.list_objects_v2(
-        Bucket=bucket,
-        Prefix=os.path.join(container_folder, ""),
-        ExpectedBucketOwner=providers.get_aws_account_id(),
-    )
-    keys = set(os.path.dirname(k.get("Key")) for k in rsp.get("Contents"))
-    prefix = rsp.get("Prefix")
-    partition_keys = [p[len(prefix) :] for p in keys if p != prefix.rstrip("/")]
+    rsp = s3.list_objects_v2(Bucket=bucket, Prefix=os.path.join(container_folder, ''))
```
Comment (w0rks74710n) [inline driver/aws/datalake_api.py:45]: `providers.get_aws_account_id()` is gone. If this is intended, I am okay with it. Just FYI.

File: tests/test_model_compilation.py [hunk 2]
```diff
@@ -21,41 +21,26 @@
...
-        raise AssertionError
-    if not (len(product.pipeline.tasks) == 1):
-        raise AssertionError
-    if not (product.pipeline.tasks[0].id == "process_some_files"):
-        raise AssertionError
-    if not (len(product.pipeline.tasks[0].inputs) == 1):
-        raise AssertionError
-    if not (len(product.pipeline.tasks[0].outputs) == 1):
-        raise AssertionError
+    assert product.engine == 'glue'
```
Comment (w0rks74710n) [inline tests/test_model_compilation.py:24]: Nice :D

**Review by w0rks74710n — DISMISSED** [review]
LGTM! Just the comment around the Account ID missing in `ExpectedBucketOwner`.

**Comment by thingsplode** [comment]
error handling and a few cleanups.

**Pull request merged**